app/src/main/java/com/zulip/android/ZulipActivity.java

```diff
@@ -54,6 +57,48 @@
         MessageListFragment.Listener {
 
     ZulipApp app;
+    List<Message> muted_topics;
+    @Override
+    public void addToList(Message message) {
+        muted_topics.add(message);
+    }
+
+    @Override
+    public void unMuteTopic(int StreamID, String subject) {
+        app.UnMuteTopic(StreamID, subject);
+        int added = 0;
+        for (Iterator<Message> messageIterator = muted_topics.iterator(); messageIterator.hasNext(); ) {
+            Message message = messageIterator.next();
+            if (message.getStream().getId() == StreamID && message.getSubject().equals(subject)) {
+                messageIterator.remove();
+                homeList.adapter.add(message);
+                added++;
+            }
+        }
+        if (added > 0) {
+            Collections.sort(homeList.messageList, new Comparator<Message>() {
```

Comment:
This might be nicer to have `Message` implement the Comparable interface and move the logic into the `Message` class.